src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -706,6 +717,7 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
 
         if self._user_schema:
             raise ValueError(f"Read {format} does not support user schema")
+        path = _validate_stage_path(path)
```

Comment:
do we need this change for other format like json, avro, parquet methods

Reply:
Those methods call _read_semi_structured_file. I've updated tests to show that they are also affected.